lwgenerate/aws_eks_audit/aws_eks_audit.go

```diff
@@ -0,0 +1,612 @@
+package aws_eks_audit
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/hashicorp/hcl/v2/hclwrite"
+	"github.com/lacework/go-sdk/lwgenerate"
+	"github.com/pkg/errors"
+	"github.com/zclconf/go-cty/cty"
+)
+
+type ExistingCrossAccountIamRoleDetails struct {
+	// Existing IAM Role ARN
+	Arn string
+
+	// Existing IAM Role External ID
+	ExternalId string
+}
+
+func (e *ExistingCrossAccountIamRoleDetails) IsPartial() bool {
+	// If nil, return false
+	if e == nil {
+		return false
+	}
+
+	// If all values are empty, return false
+	if e.Arn == "" && e.ExternalId == "" {
+		return false
+	}
+
+	// If all values are populated, return false
+	if e.Arn != "" && e.ExternalId != "" {
+		return false
+	}
+
+	return true
+}
+
+// NewExistingCrossAccountIamRoleDetails Create new existing IAM role details
+func NewExistingCrossAccountIamRoleDetails(arn string, externalId string) *ExistingCrossAccountIamRoleDetails {
+	return &ExistingCrossAccountIamRoleDetails{
+		Arn:        arn,
+		ExternalId: externalId,
+	}
+}
+
+type GenerateAwsEksAuditTfConfigurationArgs struct {
+
+	// Supply an AWS Profile name
+	AwsProfile string
+
+	// Should we require MFA for object deletion?
+	BucketEnableMfaDelete bool
+
+	// Should we disable bucket encryption?
+	BucketDisableEncryption bool
```

Comment:
This comment isn't requesting any changes.  Just opinionated concern.  I think that inverting the name and meaning, from `bucket_encryption_enabled default true` to `disable_encryption_s3 default true` is going to lead to confusion and maybe bugs.  At very least it forcing unnecessary cognitive load on users and maintainers.

I had to look in code to be sure that this option mapped to the Terraform input var.  I don't think it is clear enough.

```
--disable_encryption_s3                     disable encryption on s3 bucket (default true)
```

https://github.com/lacework/terraform-aws-eks-audit-log

bucket_encryption_enabled | Set this to true to enable encryption on a created S3 bucket | bool | true
-- | -- | -- | --

-----

As an option - I used `--flag=(true|false)` as a way of handling default true TF input.

Reply:
Understood. My reasoning for this was that in most cases in our Terraform we want to keep it as simple as possible. IMO if the user wants to set an input value equal to the Terraform input default value we should probably just not pass the input at all. Inverting the name of the input allowed me to do this.

How about I add comments to be explicit and reference which TF input these map to?

Reply:
`--bucket_encryption_enabled` default true and `--bucket_encryption_enabled=false` to disable, I think that is just as simple.

Why have a comment explaining the mapping when we could instead just have the names match up?  Self documenting.  With comments there is a hope that user has energy and focus to spend on reading them, and hope that they are actually able to find them.  Attention as resource, masses of information to filter/parse, etc.

Said my piece :D

Reply:
My view was that in almost all cases the end users are likely to be running this without actually checking the documentation on the TF registry, as if they were at the point of reading that they would likely just create the TF themselves.

I shall go the route of making this a string var